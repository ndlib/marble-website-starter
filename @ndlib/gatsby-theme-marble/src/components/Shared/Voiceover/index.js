/** @jsx jsx */
// eslint-disable-next-line no-unused-vars
import React from 'react'
import PropTypes from 'prop-types'
import { jsx } from 'theme-ui'

const active = {
  color: '#fff',
  backgroundColor: '#000',
  left: 'auto',
  top: 'auto',
  width: '30%',
  height: 'auto',
  overflow: 'auto',
  margin: '10px 35%',
  padding: '5px',
  textAlign: 'center',
  fontSize: '1.2em',
  zIndex: '999',
}

export const sx = {
  left: '-999px',
  position: 'absolute',
  top: 'auto',
  width: '1px',
  height: '1px',
  overflow: 'hidden',
  zIndex: '-999',
  '&:focus': active,
  '&:active': active,
}

const Voiceover = ({ text }) => {
  return (
    <p sx={sx} id='hiddenVoiceover'>{text}</p>
  )
}

Voiceover.propTypes = {
  text: PropTypes.object.isRequired,
}

export default Voiceover