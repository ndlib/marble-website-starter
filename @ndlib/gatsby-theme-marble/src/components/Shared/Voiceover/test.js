import React from 'react'
import renderer from 'react-test-renderer'

import Voiceover from './'

test('Voiceover', () => {
  const tree = renderer
    .create(<Voiceover text='Test' />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
