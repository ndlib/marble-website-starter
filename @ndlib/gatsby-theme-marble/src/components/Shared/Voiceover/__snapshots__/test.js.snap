// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Voiceover 1`] = `
<p
  className="css-cxwvu9-Voiceover"
  id="hiddenVoiceover"
>
  Test
</p>
`;
